cmd/whitesourceExecuteScan.go

```diff
@@ -403,60 +837,82 @@ func blockUntilProjectIsUpdated(config *ScanOptions, sys whitesource, currentTim
 }
 
 // downloadReports downloads a project's risk and vulnerability reports
-func downloadReports(config *ScanOptions, utils whitesourceUtils, sys whitesource) ([]piperutils.Path, error) {
+func downloadReports(config *ScanOptions, scan *whitesourceScan, utils whitesourceUtils, sys whitesource) ([]piperutils.Path, error) {
```

Comment:
What about extracting this and related functions to an own package `pkg/whitesource/report`?